packages/core/src/types.ts

```diff
@@ -1854,21 +1863,36 @@ export type EventOfMachine<
 export interface ActorContext<TEvent extends EventObject, TSnapshot> {
   self: ActorRef<TEvent, TSnapshot>;
   name: string;
-  _event: SCXML.Event<TEvent> | LifecycleSignal;
+  sessionId: string;
+  logger: (...args: any[]) => void;
+  exec: (fn: () => void) => void;
+  defer: ((fn: () => void) => void) | undefined;
+  observers: Set<Observer<TSnapshot>>;
```

Comment:
Does this have to be publicly exposed? It seems to be only used in the `Interpreter` and I don't see how access to this would be helpful to the users - this feels like a state that should be protected.

```suggestion
```

Reply:
Created https://github.com/statelyai/xstate/pull/3722 to address this